app/component/map/MapWithTracking.js

```diff
@@ -72,102 +72,120 @@ const onlyUpdateCoordChanges = onlyUpdateForKeys(
   ],
 );
 
-const MapWithTracking = withMapStateTracking(
-  connectToStores(
-    onlyUpdateCoordChanges(Map),
-    ['PositionStore', 'EndpointStore', 'PreferencesStore'],
-    (context, props) => {
-      const { mapTracking } = props.mapState;
-      const PositionStore = context.getStore('PositionStore');
-      const position = PositionStore.getLocationState();
-      const origin = context.getStore('EndpointStore').getOrigin();
-      const language = context.getStore('PreferencesStore').getLanguage();
-
-      let location = (() => {
-        if (props.mapState.focusOnOrigin && !origin.useCurrentPosition) {
-          return origin;
-        } else if (mapTracking && position.hasLocation) {
-          return position;
-        }
-        return false;
-      })();
-
-      if (
-        !origin.useCurrentPosition &&
-        origin !== props.mapState.previousOrigin
-      ) {
-        setTimeout(props.dispatch, 0, {
-          type: 'useOrigin',
-          origin,
+const MapWithTracking = connectToStores(
+  onlyUpdateCoordChanges(Map),
+  ['PositionStore', 'EndpointStore', 'PreferencesStore'],
+  (context, props) => {
+    const { mapTracking } = props.mapState;
+    const PositionStore = context.getStore('PositionStore');
+    const position = PositionStore.getLocationState();
+    const origin = context.getStore('EndpointStore').getOrigin();
+    const language = context.getStore('PreferencesStore').getLanguage();
+
+    let location;
+    let zoom;
+
+    console.log(context);
+
+    if (
+      props.mapState.focusOnOrigin &&
+      !origin.useCurrentPosition &&
+      origin.lat != null &&
+      origin.lon != null
+    ) {
+      location = origin;
+      zoom = 16;
+    } else if (mapTracking && position.hasLocation) {
+      location = position;
+      zoom = 16;
+    } else {
+      location = context.config.defaultMapCenter;
```

Comment:
Could we use here defaultEndpoint if map center is not defined? It could be the right choice for most configurations.